TestProviders/ReadyProvider.psm1

```diff
@@ -54,8 +54,9 @@ Class ReadyProvider : TestProvider
 			$count = 0
 			foreach ($vmData in $AllVMData) {
 				# get the first active nic device's IPv4 address. This is the temporary approach, as it may not be right for VMs that have multi nic devices.
+				# For devices with multiple interfaces, we ignore the interface present in staticipconfig.log
 				$ipAddrInfo = Run-LinuxCmd -username $global:user -password $global:password -ip $($vmData.PublicIp) -port $($vmData.SSHPort) `
-					-command "ip -4 address | awk -F': ' '!/lo/ {print `$2}' | xargs ip address show" -RunAsSudo
+					-command "interface=`grep -s 'Available port' /tmp/staticipconfig.log | awk -F': ' '{print $2}'`; ip -4 address | awk -v var=`"$interface`" -F': ' '!/lo/ { if ($2!=var) print $2}' | xargs ip address show" -RunAsSudo
```

Comment:
when and where is the /tmp/staticipconfig.log generated?

Reply:
This file contains name of interfaces which are not currently up. I discussed with Johnson and I will move this additional logic to an if/else block to make it clear that this is not part of normal workflow.